users/userservice/user-service.js

```diff
@@ -43,7 +43,7 @@ function validateUsername(username) {
   
   // Allow only letters, numbers, and underscores, and ensure length between 3 and 20 characters
   const usernameRegex = /^[a-zA-Z0-9_]{3,20}$/;
-  if (!usernameRegex.test(usernameStr)) {
+  if (username==="" || !usernameRegex.test(usernameStr)) {
```

Comment:
I mean the `regex` validation also covers the empty string, although it still improves code readability i don't think this is necessary. I would leave it as is.
```suggestion
  if (!usernameRegex.test(usernameStr)) {
```